lib/object.c

```diff
@@ -201,13 +201,12 @@ static tool_rc tpm2_util_object_fetch_parent_from_tpk(const char *objectstr,
         goto ret;
     }
 
-    uint64_t parent;
-    int ret = ASN1_INTEGER_get_uint64(&parent, tpk->parent);
-    if (ret == 0 || parent > UINT32_MAX) {
+    TPM2_HANDLE ret = ASN1_INTEGER_get(tpk->parent);
```

Comment:
This returns a long, which is signed, which then get converted to the unsigned TPM2_HANDLE, which would be a weird handle. also the man page states that, "The deprecated function [ASN1_INTEGER_get](https://man.openbsd.org/ASN1_INTEGER_get.3#ASN1_INTEGER_get)() converts a to the long type."

I remember their being some issues with negative handles in tpm2-tss-engine code, see:
  - https://github.com/tpm2-software/tpm2-tss-engine/pull/222

It't important to note that they saved a pem file for testing that has the old negative handle in it to ensure backwards compatibility. We should do the same thing here, they use the BN routines and then BN to ASN1 and also check for negative if it's negative use ASN1_INTEGER_get and if positive use BN_get_word. We still probably want to use ASN1_INTEGER_get_int64 or something because of the deprecation notes in the man pages (may be OSSL version dependent).

Reply:
```suggestion
    BIGNUM *bn_parent =  ASN1_INTEGER_to_BN(tpk->parent, NULL);
    if (!bn_parent) {
         goto error;
    }
    
    TPM2_HANDLE ret = 0;
    bool is_bn_parent_negative = BN_is_negative(bn_parent);
    if (is_bn_parent_negative) {
         ret = ASN1_INTEGER_get(tpk->parent);
     } else {
         ret = BN_get_word(bn_parent);
     }
```